src/battle_controller_player.c

```diff
@@ -1889,6 +1889,19 @@ static void HandleChooseActionAfterDma3(u32 battler)
     {
         gBattle_BG0_X = 0;
         gBattle_BG0_Y = DISPLAY_HEIGHT;
+        if (gBattleStruct->aiDelayTimer != 0)
+        {
+            gBattleStruct->aiDelayFrames = gMain.vblankCounter1 - gBattleStruct->aiDelayTimer;
+            gBattleStruct->aiDelayTimer = 0;
+            #if DEBUG_AI_DELAY_TIMER
+            {
+                static const u8 sText_AIDelay[] = _("AI delay:\n{B_BUFF1} frames");
```

Comment:
does declaring this here put it on the stack? If so, might be better to define it outside of the function

Reply:
If you declare a variable as `static` inside a function then it's basically a global that can't be accessed outside of the function (so not on the stack), and then additionally because it's `const` it goes into `.rodata`/ROM rather than RAM.

Reply:
It would be put on stack if I didn't add `static`, as mrgriffin said.